sxs/simulations/analyze.py

```diff
@@ -0,0 +1,265 @@
+import multiprocessing
+
+from ..waveforms.norms import create_unified_waveforms, compute_L2_norm, compute_mismatch
+from ..waveforms.alignment import align_waveforms, align_simulations
+from ..handlers import load
+
+import numpy as np
+
+
+def compute_error_summary(wa, wb, t1, t2, ASDs=None, total_masses=None):
+    """
+    Compute various errors between two waveforms.
+
+    This computes the time-domain mismatch over the two-sphere,
+    the normalized L² norm of the residual, the frequency-domain mismatch
+    against whatever detector ASDs are provided with some total mass,
+    and the individual modes' residual norms and norms.
+
+    Parameters
+    ----------
+    wa : WaveformModes
+    wb : WaveformModes
+    t1 : float
+        Beginning of integrals.
+    t2 : float
+        End of integrals.
+    ASDs : dict of funcs, optional
+        Dictionary of functions mapping frequencies to the ASD of a detector(s).
+        Default is no frequency-domain mismatch is calculated.
+    total_masses : list of floats, optional
+        Total masses in solar masses to use for frequency-domain mismatches.
+        Default is 1.
+
+    Returns
+    -------
+    errors : dict
+        Dictionary of the time-domain mismatch over the two-sphere,
+        the normalized L² norm of the residual, the frequency-domain mismatch
+        against whatever detector ASDs are provided with some total mass,
+        and the modes' absolute errors and norms.
+    """
+    from .. import m_sun_in_seconds
+    
+    errors = {}
+
+    errors["t1"] = t1
+    errors["t2"] = t2
+
+    wa, wb = create_unified_waveforms(wa, wb, t1, t2, padding_time_factor=0)
+
+    errors["mismatch"] = compute_mismatch(wa, wb, t1, t2)
+
+    errors["residual L2 norm"] = compute_L2_norm(wa, wb, t1, t2)
+
+    f1 = (np.gradient(-np.unwrap(np.angle(wa.data[:, wa.index(2, 2)])), np.diff(wa.t * m_sun_in_seconds)[0]))[
+        np.argmin(abs(wa.t - t1))
+    ] / (2 * np.pi)
```

Comment:
The more I look at this, the more I'm skeptical of it.  If there's noise or precession, this can go wonky.  A more robust approach is to just use the angular_velocity.  Something like this (with units adjusted appropriately; see other comments) would make me happier:
```suggestion
    i1 = wa.index_closest_to(t1)
    i0, i2 = max(0, i1-5), min(i1+6, wa.n_times-1)
    Ω1 = np.linalg.norm(h[i0:i2].angular_velocity, axis=1)[5]
    f1 = 2*Ω1 / (2*np.pi)
```